notebooks/official/workbench/spark/spark_sample_notebook.ipynb

```diff
@@ -0,0 +1,1200 @@
+    "\n",
+    "**Note**: Jupyter runs lines prefixed with `!` as shell commands, and inserts the value of Python variables prefixed with `$` into the commands."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "lWEdiXsJg0XY"
+   },
+   "source": [
+    "### Create a Dataproc cluster\n",
+    "\n",
+    "The Spark job executed in this notebook tutorial is compute intensive. Since the job can take a significant amount time to complete in a standard notebook environment, this notebook tutorial runs on a Dataproc cluster that is created with the Dataproc Component Gateway and Jupyter component installed on the cluster.\n",
+    "\n",
+    "**Existing Dataproc with Jupyter cluster?**: If you have a running Dataproc cluster you can use for this tutorial (it must have the [Component Gateway and Jupyter component installed on the cluster](https://cloud.google.com/dataproc/docs/concepts/components/jupyter#gcloud-command)), you can skip this and go to `Switch your kernel`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "oM1iC_MfAts1"
+   },
+   "outputs": [],
+   "source": [
+    "CLUSTER_NAME = \"[your-cluster]\"  # @param {type: \"string\"}\n",
+    "CLUSTER_REGION = \"[your-region]\"  # @param {type: \"string\"}\n",
+    "\n",
+    "if CLUSTER_REGION == \"[your-region]\":\n",
+    "    CLUSTER_REGION = \"us-central1\"\n",
+    "\n",
+    "print(f\"CLUSTER_NAME: {CLUSTER_NAME}\")\n",
+    "print(f\"CLUSTER_REGION: {CLUSTER_REGION}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "oM1iC_MfAts1"
+   },
+   "outputs": [],
+   "source": [
+    "!gcloud dataproc clusters create $CLUSTER_NAME \\\n",
+    "    --region=$CLUSTER_REGION \\\n",
+    "    --enable-component-gateway \\\n",
+    "    --optional-components=JUPYTER"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "lWEdiXsJg0XY"
+   },
+   "source": [
+    "Your `CLUSTER_NAME` must be **unique within your Google Cloud project**. It must start with a lowercase letter, followed by up to 51 lowercase letters, numbers, and hyphens, and cannot end with a hyphen."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "lWEdiXsJg0XY"
+   },
+   "source": [
+    "#### Switch your kernel\n",
+    "\n",
+    "Your notebook kernel is listed at the top of the notebook page. Your notebook should run on the Python 3 kernel running on your Dataproc cluster.\n",
+    "\n",
+    "Select **Kernel > Change Kernel** from the top menu, then select `Python 3 on CLUSTER_NAME: Dataproc cluster in REGION (Remote)`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "WReHDGG5g0XY"
+   },
+   "source": [
+    "### Set your project ID\n",
+    "\n",
+    "Run the following cell to try to get you project ID."
```

Comment:
Run the following cell to try to get you project ID -> Run the following cell to get you project ID